src/JDbg/Jdwp.cs

```diff
@@ -66,21 +66,26 @@ private void OnDisconnect(/*OPTIONAL*/ SocketException socketException)
             // This handles disconnects if, for example, the app exits
 
             // If we have any waiting operations, we want to abort them
+            var operationsToAbort = new List<WaitingOperationDescriptor>();
             lock (_waitingOperations)
             {
                 foreach (var operation in _waitingOperations)
                 {
-                    if (socketException == null)
-                    {
-                        operation.Value.Abort();
-                    }
-                    else
-                    {
-                        operation.Value.OnSocketError(socketException);
-                    }
+                    operationsToAbort.Add(operation.Value);
```

Comment:
nit: use operationsToAbort = new List<WaitingOperationDescriptor>(_waitingOperations.Values);

So that List can be smart about settings its size based on the size of the collection being copied.